ibis/backends/tests/test_vectorized_udf.py

```diff
@@ -336,6 +339,45 @@ def test_elementwise_udf_overwrite_destruct_and_assign(backend, alltypes):
     backend.assert_frame_equal(result, expected, check_like=True)
 
 
+@pytest.mark.only_on_backends(['pandas', 'pyspark'])
+# TODO - udf - #2553
+@pytest.mark.xfail_backends(['dask'])
+@pytest.mark.xfail_unsupported
+def test_elementwise_udf_destruct_exact_once(backend, alltypes):
+    # xfail on Spark version < 3.1.1
+    try:
+        from distutils.version import LooseVersion
+
+        import pyspark
+
+        if LooseVersion(pyspark.__version__) < LooseVersion("3.1.1"):
+            pytest.xfail("This test only works on Spark >= 3.1.1")
+    except ImportError:
+        pass
+
+    with tempfile.TemporaryDirectory() as tempdir:
+
+        @elementwise(
+            input_type=[dt.double],
+            output_type=dt.Struct(['col1', 'col2'], [dt.double, dt.double]),
+        )
+        def add_one_struct_exact_once(v):
+            print(v)
+            key = v.iloc[0]
+            path = Path(f"{tempdir}/{key}")
+            assert not path.exists()
```

Comment:
why are you writing things?

Reply:
This is basically to create a side effect. If this function gets run for the second time it will hit the side effect and fail.

Reply:
ok cool (may want to document this in the future)